web/src/unread_ops.ts

```diff
@@ -360,6 +411,10 @@ function do_mark_unread_by_narrow(
                 );
             } else if (loading_indicator_displayed) {
                 finish_loading(messages_marked_unread_till_now);
+                if (unsubscribed_ignored_channels.length > 0) {
+                    handle_skipped_unsubscribed_streams(unsubscribed_ignored_channels);
+                    unsubscribed_ignored_channels = [];
+                }
```

Comment:
Why does this only need to happen if `loading_indicator_displayed`? 

If it does, I expect it can be refactored into being part of `finish_loading` ... but I don't see why we can assume that a single-batch call would not end up here.